sdk/program/src/vote/state/mod.rs

```diff
@@ -479,13 +479,43 @@ impl VoteState {
         }
     }
 
-    /// Deserializes the input `VoteStateVersions` buffer directly into a provided `VoteState` struct
+    /// Deserializes the input `VoteStateVersions` buffer directly into the provided `VoteState`.
     ///
-    /// In a BPF context, V0_23_5 is not supported, but in non-BPF, all versions are supported for
-    /// compatibility with `bincode::deserialize`
+    /// In a SBPF context, V0_23_5 is not supported, but in non-SBPF, all versions are supported for
+    /// compatibility with `bincode::deserialize`.
     pub fn deserialize_into(
         input: &[u8],
         vote_state: &mut VoteState,
+    ) -> Result<(), InstructionError> {
+        // Safety: vote_state is valid to_drop (see drop_in_place() docs). After
+        // dropping, the pointer is treated as uninitialized and only accessed
+        // through ptr::write, which is safe as per drop_in_place docs.
+        unsafe {
+            std::ptr::drop_in_place(vote_state);
+        }
+
+        VoteState::deserialize_into_ptr(input, vote_state)
+    }
+
+    /// Deserializes the input `VoteStateVersions` buffer directly into the provided
+    /// `MaybeUninit<VoteState>`.
+    ///
+    /// In a SBPF context, V0_23_5 is not supported, but in non-SBPF, all versions are supported for
+    /// compatibility with `bincode::deserialize`.
+    ///
+    /// On success, `vote_state` is fully initialized and can be converted to `VoteState` using
+    /// [MaybeUninit::assume_init]. On failure, `vote_state` may still be uninitialized and must not
+    /// be converted to `VoteState`.
+    pub fn deserialize_into_uninit(
+        input: &[u8],
+        vote_state: &mut MaybeUninit<VoteState>,
+    ) -> Result<(), InstructionError> {
+        VoteState::deserialize_into_ptr(input, vote_state.as_mut_ptr())
```

Comment:
nit: `Self::deserialzie_into_ptr()`? (i.e. can use `Self`).

Reply:
There are other pre-existing bits that already use VoteState instead of Self, so I'll leave it to VoteState (which also happens to be my preference :P)